tsdate/evaluation.py

```diff
@@ -252,3 +252,107 @@ def match_node_ages(ts, other):
     matched_time[matched_span == 0] = np.nan
 
     return matched_time, matched_span, best_match
+
+
+def node_spans(ts):
+    """
+    Returns the array of "node spans", i.e., the `j`th entry gives
+    the total span over which node `j` is in the tree (i.e., does
+    not have 'missing data' there).
+    """
+    child_spans = np.bincount(
+        ts.edges_child,
+        weights=ts.edges_right - ts.edges_left,
+        minlength=ts.num_nodes,
+    )
+    for t in ts.trees():
+        span = t.span
+        for r in t.roots:
+            # do this check to exempt 'missing data'
+            if t.num_children(r) > 0:
+                child_spans[r] += span
+    return child_spans
+
+
+def tree_discrepancy(ts, other):
+    """
+    For two tree sequences `ts` and `other`,
+    this method `tree_discrepancy` returns two values:
+    1. the sum across the nodes of `ts`
+    of the length of the discrepancy in span between the node
+    and its best match node in `other`
+    weighted by difference in time.
```

Comment:
```suggestion
    1. the fraction of the total span of `ts` over which each nodes' descendant
    sample set does not match its' best match's descendant sample set.
```
and perhaps say later on
```
    The first value is computed as `A/B`, where `B` is the sum of the spans
    of all nodes in `ts`, and `A` is the
    sum across the nodes of `ts` of the length of the span
    over which the node and its best match node in `other` do not have the
    same set of descendant samples
    weighted by difference in time.
```